compiler/src/dotty/tools/dotc/core/CheckRealizable.scala

```diff
@@ -96,6 +96,14 @@ class CheckRealizable(implicit ctx: Context) {
       else boundsRealizability(tp).andAlso(memberRealizability(tp))
   }
 
+  private def refinedNames(tp: Type): Set[Name] = tp.dealias match {
+    case tp: RefinedType => refinedNames(tp.parent) + tp.refinedName
+    case tp: AndType => refinedNames(tp.tp1) ++ refinedNames(tp.tp2)
+    case tp: OrType  => refinedNames(tp.tp1) ++ refinedNames(tp.tp2)
```

Comment:
It seems that `A | B` would need a test? Confused what happens there with disjoint ranges.

Reply:
`refinedNames` only collects names that appear in a top-level refinement. So that's the same for & and |. The bounds checking is done later.

Reply:
I meant I don't see a testcase exercising this path.